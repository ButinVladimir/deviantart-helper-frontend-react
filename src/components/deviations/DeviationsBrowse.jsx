import React from 'react';
import PropTypes from 'prop-types';
import DeviationsBrowseForm from './DeviationsBrowseFormContainer';
import DeviationsBrowseList from './DeviationsBrowseListContainer';
import Config from '../../config/config';

export default function DeviationsBrowse({
  config,
}) {
  return (
    <div>
      <DeviationsBrowseForm config={config} />
      <DeviationsBrowseList config={config} />
    </div>
  );
}

DeviationsBrowse.propTypes = {
  config: PropTypes.instanceOf(Config).isRequired,
};
