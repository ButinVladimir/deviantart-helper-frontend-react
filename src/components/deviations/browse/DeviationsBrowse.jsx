import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import DeviationsBrowseForm from './DeviationsBrowseFormContainer';
import DeviationsBrowseList from './DeviationsBrowseListContainer';
import DeviationsBrowsePagination from './DeviationsBrowsePaginationContainer';
import Config from '../../../config/config';

export default class DeviationsBrowse extends Component {
  componentDidMount() {
    const { clearDataHander } = this.props;
    clearDataHander();
  }

  render() {
    const { config } = this.props;

    return (
      <Fragment>
        <DeviationsBrowseForm config={config} />
        <DeviationsBrowseList config={config} />
        <DeviationsBrowsePagination config={config} />
      </Fragment>
    );
  }
}

DeviationsBrowse.propTypes = {
  config: PropTypes.instanceOf(Config).isRequired,
  clearDataHander: PropTypes.func.isRequired,
};
