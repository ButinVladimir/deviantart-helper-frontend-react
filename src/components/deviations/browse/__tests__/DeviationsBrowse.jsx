import React from 'react';
import { shallow } from 'enzyme';
import DeviationsBrowse from '../DeviationsBrowse';

describe('DeviationsBrowse', () => {
  it('can be rendered correctly when page is not loading', () => {
    const wrapper = shallow(
      <DeviationsBrowse
        pageLoading={false}
        clearDataHandler={() => {}}
        preloadDeviationsHandler={() => {}}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('can be rendered correctly when page is loading', () => {
    const wrapper = shallow(
      <DeviationsBrowse
        pageLoading
        clearDataHandler={() => {}}
        preloadDeviationsHandler={() => {}}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('handles clearing data', () => {
    const clearDataHandler = jest.fn();
    shallow(
      <DeviationsBrowse
        pageLoading
        clearDataHandler={clearDataHandler}
        preloadDeviationsHandler={() => {}}
      />,
    );

    expect(clearDataHandler).toHaveBeenCalled();
  });

  it('handles preloading deviations', () => {
    const preloadDeviationsHandler = jest.fn();
    shallow(
      <DeviationsBrowse
        pageLoading
        clearDataHandler={() => {}}
        preloadDeviationsHandler={preloadDeviationsHandler}
      />,
    );

    expect(preloadDeviationsHandler).toHaveBeenCalled();
  });
});
