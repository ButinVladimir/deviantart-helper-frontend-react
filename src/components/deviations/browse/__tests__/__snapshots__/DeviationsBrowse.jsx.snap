// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviationsBrowse can be rendered correctly when page is loading 1`] = `
<Fragment>
  <Component />
  <CustomLoader />
  <Component />
</Fragment>
`;

exports[`DeviationsBrowse can be rendered correctly when page is not loading 1`] = `
<Fragment>
  <Component />
  <Connect(DeviationsBrowseList) />
  <Component />
</Fragment>
`;
